certbot-apache/certbot_apache/http_01.py

```diff
@@ -105,8 +100,10 @@ def _set_up_challenge(self, achall):
         response, validation = achall.response_and_validation()
 
         name = os.path.join(self.challenge_dir, achall.chall.encode("token"))
+
         with open(name, 'wb') as f:
             f.write(validation.encode())
+            self.configurator.reverter.register_file_creation(True, name)
```

Comment:
We should register the file with the reverter before creating it. While unlikely, if we get a signal/exception between writing the file and finishing registering it, it will never be deleted.